### PR TITLE
sdk.models.application: getAllWithDeviceServiceDetails method added

### PR DESCRIPTION
**Pull request by karaxuna**
Resolves: #610
Change-type: minor

### CODE REVIEW

File: lib/models/application.coffee [hunk 1]
```diff
@@ -130,6 +130,51 @@ getApplicationModel = (deps, opts) ->
 
 		.asCallback(callback)
 
+	###*
+	# @summary Get applications and their deives, along with each device's
```
Comment (sradevski) [inline lib/models/application.coffee]: NIT: Misspelled word, "their deives" -> "their devices".

**Comment by karaxuna** [comment]
@thgreasi can we create another issue for these tests? This issue is blocking me on https://github.com/balena-io/resin-ui/issues/2422 and the last one blocks me on https://github.com/balena-io/resin-ui/pull/2490. Writing tests for this issue will take some time for me considering the fact that I'm not that familiar with sdk repo.

**Comment by thgreasi** [comment]
@karaxuna You should be able to start the balena sdk upgrade in the UI w/o this PR.
Perhaps you can write the tests for this while we will be reviewing the sdk upgrade PR.
If the tests end up blocking you, then we can probably open an SDK for them and merge this a bit sooner, as far as the new methods works fine in the UI PR.

File: lib/models/application.coffee [hunk 2]
```diff
@@ -130,6 +130,51 @@ getApplicationModel = (deps, opts) ->
...
+
+		serviceOptions = mergePineOptions
+			$expand: [
+				owns__device: getCurrentServiceDetailsPineOptions()
+			]
+		, options
+
+		exports.getAll(serviceOptions)
+		.then (apps) ->
+			return apps.map (app) ->
```
Comment (thgreasi) [inline lib/models/application.coffee]: NIT: we probably don't need to use `.map` here. Iterating with `forEach` to set the `owns__device` property on the existing objects and returning the `apps` should be fine and would save an unnecessary array creation.

**Comment by karaxuna** [comment]
@resin-ci retest

**Pull request merged**